x-pack/plugin/sql/sql-proto/src/test/java/org/elasticsearch/xpack/sql/proto/SqlVersionTests.java

```diff
@@ -73,4 +75,34 @@ public void testVersionsAndInvalidStringNotEqual() {
         assertNotEquals(ver1, ver2);
     }
 
+    public void testVersionCompatibilityClientWithNoCompatibility() {
+        SqlVersion server = SqlVersion.fromId(CURRENT.id);
+        int major = randomIntBetween(1, 7);
+        SqlVersion client = new SqlVersion(major, randomIntBetween(0, major == 7 ? 6 : 99), randomIntBetween(0, 99));
+        assertFalse(isClientCompatible(server, client));
+    }
+
+    public void testVersionCompatibilityClientNewer() {
+        int major = randomIntBetween(7, 99);
+        SqlVersion server = new SqlVersion(major, randomIntBetween(major > 7 ? 0 : 7, 99), randomIntBetween(0, 98));
+        SqlVersion client = new SqlVersion(server.major, server.minor, (byte) (server.revision + 1));
+        assertFalse(isClientCompatible(server, client));
+    }
+
+    public void testVersionCompatibilityClientTooOld() {
+        int major = randomIntBetween(9, 99);
+        SqlVersion server = new SqlVersion(major, randomIntBetween(major > 7 ? 0 : 7, 99), randomIntBetween(0, 98));
```

Comment:
I don't think that the `major > 7 ? 0 : 7, 99` is needed since client is always `major - 2`, seems confusing, unless I'm missing something.

Reply:
Right, the condition being actually always true. Fixed, thanks. (c&p from above test)